### PR TITLE
erofs-utils: enable lzma support

### PR DESCRIPTION
**Pull request by hsiangkao**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/e/erofs-utils.rb [hunk 1]
```diff
@@ -24,6 +24,7 @@ class ErofsUtils < Formula
   depends_on "pkg-config" => :build
   depends_on "lz4"
   depends_on "util-linux" # for libuuid
+  depends_on "xz"
```
Comment (ZhongRuoyu) [inline Formula/e/erofs-utils.rb:27]: macOS provides `xz` (liblzma). Can that be used?

If so, we can use

```ruby
uses_from_macos "xz"
```

Comment (hsiangkao) [reply Formula/e/erofs-utils.rb:27]: erofs-utils needs to use xz >= 5.4. I'm not sure if the native macOS liblzma version is enough.

Comment (ZhongRuoyu) [reply Formula/e/erofs-utils.rb:27]: That's very new. Okay, let's use the `xz` formula then.

**Review by ZhongRuoyu — APPROVED** [review]
Thanks @hsiangkao!

**Pull request merged**